pavelib/quality.py

```diff
@@ -1,203 +1,97 @@
+    Run Pylint on system code, only looking for FIXME items
     """
-    num_fixme = 0
-    systems = getattr(options, 'system', ALL_SYSTEMS).split(',')
-
-    for system in systems:
-        # Directory to put the pylint report in.
-        # This makes the folder if it doesn't already exist.
-        report_dir = (Env.REPORT_DIR / system).makedirs_p()
-
-        apps = [system]
-
-        for directory in ['djangoapps', 'lib']:
-            dirs = os.listdir(os.path.join(system, directory))
-            apps.extend([d for d in dirs if os.path.isdir(os.path.join(system, directory, d))])
-
-        apps_list = ' '.join(apps)
-
-        pythonpath_prefix = (
-            "PYTHONPATH={system}:{system}/lib"
-            "common/djangoapps:common/lib".format(
-                system=system
-            )
-        )
-
-        sh(
-            "{pythonpath_prefix} pylint --disable R,C,W,E --enable=fixme "
-            "--msg-template={msg_template} {apps} "
-            "| tee {report_dir}/pylint_fixme.report".format(
-                pythonpath_prefix=pythonpath_prefix,
-                msg_template='"{path}:{line}: [{msg_id}({symbol}), {obj}] {msg}"',
-                apps=apps_list,
-                report_dir=report_dir
-            )
-        )
-
-        num_fixme += _count_pylint_violations(
-            "{report_dir}/pylint_fixme.report".format(report_dir=report_dir))
-
-    print("Number of pylint fixmes: " + str(num_fixme))
+    systems = getattr(options, 'system', ','.join(ALL_SYSTEMS)).split(',')
+    flags = [
+        '--disable R,C,W,E',
+        '--enable=fixme',
+    ]
+    pylint('fixme', systems, *flags)
 
 
 @task
 @needs('pavelib.prereqs.install_python_prereqs')
 @cmdopts([
-    ("system=", "s", "System to act on"),
-    ("errors", "e", "Check for errors only"),
-    ("limit=", "l", "limit for number of acceptable violations"),
+    ('system=', 's', 'System to act on'),
+    ('errors', 'e', 'Check for errors only'),
+    ('limit=', 'l', 'Limit for number of acceptable violations'),
 ])
 def run_pylint(options):
     """
-    Run pylint on system code. When violations limit is passed in,
+    Run Pylint on system code.
+
+    When violations limit is passed in,
     fail the task if too many violations are found.
     """
-    num_violations = 0
-    violations_limit = int(getattr(options, 'limit', -1))
+    limit = int(getattr(options, 'limit', -1))
     errors = getattr(options, 'errors', False)
-    systems = getattr(options, 'system', ALL_SYSTEMS).split(',')
-    
-    # Make sure the metrics subdirectory exists
-    Env.METRICS_DIR.makedirs_p()
-
-    for system in systems:
-        # Directory to put the pylint report in.
-        # This makes the folder if it doesn't already exist.
-        report_dir = (Env.REPORT_DIR / system).makedirs_p()
-
-        flags = []
-        if errors:
-            flags.append("--errors-only")
-
-        apps = [system]
-
-        for directory in ['lib']:
```

Comment:
`lms/djangoapps` and `cms/djangoapps` have `__init__.py` files, so they should be counted by checking `lms` and `cms` (the `system` in line 91). If you tell pylint to check the `djangoapps` dir directly, I think you end up double-counting errors.